src/backend/distributed/commands/comment.c

```diff
@@ -0,0 +1,146 @@
+/*-------------------------------------------------------------------------
+ *
+ * comment.c
+ *    Commands to interact with the comments for all database
+ *    object types.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/genam.h"
+#include "access/htup_details.h"
+#include "access/table.h"
+#include "catalog/pg_shdescription.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+#include "utils/rel.h"
+
+#include "distributed/comment.h"
+
+
+const CommentStmtType commentStmtTypes[] = {
+	{ OBJECT_DATABASE, "DATABASE" },
+	{ OBJECT_ROLE, "ROLE" }
+};
+
+
+static char * GetCommentObjectType(ObjectType objectType);
+static char * GetCommentForObject(Oid oid);
+
+
+List *
+GetCommentPropagationCommands(Oid oid, char *objectName, ObjectType objectType)
+{
+	List *commands = NIL;
+
+	StringInfo commentStmt = makeStringInfo();
+
+	/* Get the comment for the database */
+	char *comment = GetCommentForObject(oid);
+	char *commentObjectType = GetCommentObjectType(objectType);
+
+	/* Create the SQL command to propagate the comment to other nodes */
+	if (comment != NULL)
+	{
+		appendStringInfo(commentStmt, "COMMENT ON %s %s IS %s;", commentObjectType,
+						 quote_identifier(objectName),
+						 quote_literal_cstr(comment));
+	}
+
+
+	/* Add the command to the list */
+	if (commentStmt->len > 0)
+	{
+		commands = list_make1(commentStmt->data);
+	}
+
+	return commands;
+}
+
+
+static char *
+GetCommentObjectType(ObjectType objectType)
+{
+	char *objectName = NULL;
+	for (int i = 0; i < sizeof(commentStmtTypes) / sizeof(CommentStmtType); i++)
+	{
+		if (commentStmtTypes[i].objectType == objectType)
```

Comment:
If you use the approach I suggested in the other comment to store the object names then you don't need this `for` loop:

```c
const char * const ObjectTypeNames[] =
{
    [OBJECT_DATABASE] = "DATABASE",
    [OBJECT_ROLE] = "ROLE",
    [OBJECT_TSCONFIGURATION]  = "TEXT SEARCH CONFIGURATION",
    [OBJECT_TSDICTIONARY]  = "TEXT SEARCH DICTIONARY",
    /* etc. */
};
```

Then you can simply do the following to get the type name:
```c
ObjectTypeNames[OBJECT_DATABASE]
```

Reply:
Oh, hmm I now see that you did that in another place. Let's reuse that one object here too.

Reply:
Merged now